src/librbd/librbd.cc

```diff
@@ -6140,36 +6140,55 @@ extern "C" int rbd_aio_write2(rbd_image_t image, uint64_t off, size_t len,
   return 0;
 }
 
-extern "C" int rbd_aio_writev(rbd_image_t image, const struct iovec *iov,
-                              int iovcnt, uint64_t off, rbd_completion_t c)
+static int get_iovec_length(const struct iovec *iov, int iovcnt, ssize_t &len)
```

Comment:
Move these new helpers to the beginning of the file, after `create_write_raw` definition.

Reply:
Ack